### PR TITLE
feat: add kzg commitment length check when validating gossip blocks

### PR DESCRIPTION
**Pull request by ensi321**
We have missed a kzg commitment length check when validating incoming gossip block as laid out [here](https://github.com/ethereum/consensus-specs/blob/001caab3b34732420fdbddb0fff4cb50e21a84ef/specs/deneb/p2p-interface.md?plain=1#L149) in the spec.

This PR adds rejection of blocks when the number of kzg commitment is greater than `MAX_BLOBS_PER_BLOCK`

### CODE REVIEW

**Review by nflaig — DISMISSED** [review]
Looks good, seems like we only implement the [check when processing the block](https://github.com/ChainSafe/lodestar/blob/dad9037e7739d5bcbccfe627e715ef40e9ba935b/packages/state-transition/src/block/processExecutionPayload.ts#L52-L54) as per spec [here](https://github.com/ethereum/consensus-specs/blob/001caab3b34732420fdbddb0fff4cb50e21a84ef/specs/deneb/beacon-chain.md?plain=1#L375-L376).

I guess the gossip check is not as critical since we won't be importing an invalid block due to the check above. The worst that can happen is probably that we re-gossip an invalid block and get downscored by our peers.

File: packages/beacon-node/src/chain/validation/block.ts [hunk 1]
```diff
@@ -110,6 +110,18 @@ export async function validateGossipBlock(
     });
   }
 
+  // [REJECT] The length of KZG commitments is less than or equal to the limitation defined in Consensus Layer -- i.e. validate that len(body.signed_beacon_block.message.blob_kzg_commitments) <= MAX_BLOBS_PER_BLOCK
+  if (isForkBlobs(fork)) {
+    const numCommitments = (block as deneb.BeaconBlock).body.blobKzgCommitments.length;
```
Comment (nflaig) [inline packages/beacon-node/src/chain/validation/block.ts]: should consider using consistent names https://github.com/ChainSafe/lodestar/blob/dad9037e7739d5bcbccfe627e715ef40e9ba935b/packages/state-transition/src/block/processExecutionPayload.ts#L51-L52

ie. `blobKzgCommitmentsLen` vs. `numCommitments`

**Comment by twoeths** [comment]
good finding, need to update respective unit test too

**Review by twoeths — APPROVED** [review]

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.24.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.24.0) :tada:

**Pull request merged**